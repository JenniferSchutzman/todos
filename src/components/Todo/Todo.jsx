import React from 'react';

import {
  Button,
  Icon,
  Image,
  Item,
  Label,
} from 'semantic-ui-react';

const paragraph = <Image src="https://react.semantic-ui.com/images/wireframe/short-paragraph.png" />;
const Todo = () => (
  <Item>
    <Item.Content>
      <Item.Header as="a">My Neighbor Totoro</Item.Header>
      <Item.Meta>
        <span className="cinema">IFC Cinema</span>
      </Item.Meta>
      <Item.Description>{paragraph}</Item.Description>
      <Item.Extra>
        <Button primary floated="right">
          Buy tickets
          <Icon name="right chevron" />
        </Button>
        <Label>Limited</Label>
      </Item.Extra>
    </Item.Content>
  </Item>
);

export default Todo;
