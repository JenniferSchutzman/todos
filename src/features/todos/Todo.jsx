/* eslint-disable object-curly-newline */
/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import {
  Button,
  Item,
  Label,
} from 'semantic-ui-react';

import { toggleTodoDone } from './todosReducer';

const mapState = () => ({});

const mapDispatch = {
  toggleTodoDone,
};

const Todo = (props) => {
  const { due, id, labels, note, title, toggleTodoDone } = props; // eslint-disable-line no-shadow

  const handleClick = () => {
    toggleTodoDone(id);
  };

  const renderLabels = () => labels.map((label, idx) => <Label key={idx}>{label}</Label>);

  return (
    <Item>
      <Item.Content>
        <Item.Header as="a">{title}</Item.Header>
        {
          due && (
            <Item.Meta>
              <span>{`Due: ${due}`}</span>
            </Item.Meta>
          )
        }
        {
          note && (
            <Item.Description>{note}</Item.Description>
          )
        }
        <Item.Extra>
          <Button.Group floated="right">
            <Button positive onClick={handleClick}>Done</Button>
            <Button.Or />
            <Button negative>Delete</Button>
          </Button.Group>
          {
            labels && (
              <Label.Group>
                {renderLabels()}
              </Label.Group>
            )
          }
        </Item.Extra>
      </Item.Content>
    </Item>
  );
};

Todo.propTypes = {
  due: PropTypes.string,
  id: PropTypes.string.isRequired,
  labels: PropTypes.array,
  note: PropTypes.string,
  title: PropTypes.string.isRequired,
  toggleTodoDone: PropTypes.func.isRequired,
};

Todo.defaultProps = {
  due: '',
  labels: [],
  note: '',
};

export default connect(mapState, mapDispatch)(Todo);
/* eslint-enable react/no-array-index-key */
/* eslint-enable object-curly-newline */
