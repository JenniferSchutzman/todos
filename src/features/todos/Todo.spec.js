/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Todo from './Todo';

const mockStore = configureStore();
const initialState = { todos: [] };
const store = mockStore(initialState);

describe('<Todos />', () => {
  describe('render()', () => {
    test('renders the compnent', () => {
      const wrapper = shallow(<Todo store={store} id="1" title="flerm" />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
/* eslint-enable react/jsx-filename-extension */
