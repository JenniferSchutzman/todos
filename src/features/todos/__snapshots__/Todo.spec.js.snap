// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todos /> render() renders the compnent 1`] = `
<ContextProvider
  value={null}
>
  <Todo
    deleteTodo={[Function]}
    done={false}
    due=""
    labels={Array []}
    note=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    toggleTodoDone={[Function]}
  />
</ContextProvider>
`;
